py-polars/tests/parametric/test_groupby_dynamic.py

```diff
@@ -0,0 +1,217 @@
+# --------------------------------------------------------
+# Validate groupby_dynamic behaviour with parametric tests
+# --------------------------------------------------------
+#
+# Note: there are three tests here which look very similar.
+# The reason they are split into different tests is due to
+# some upstream bugs in pandas/pyarrow which would cause some tests
+# to fail if they were all in the same test function. Ideally,
+# once upstream is fixed, these tests should be merged into one.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import sys
+
+    from polars.type_aliases import StartBy
+
+    if sys.version_info >= (3, 8):
+        from typing import Literal
+    else:
+        from typing_extensions import Literal
+
+import pandas as pd
+import pytz
+from hypothesis import given, reject
+from hypothesis import strategies as st
+
+import polars as pl
+from polars.testing.parametric.strategies import strategy_time_zone_aware_series
+
+
+def _compare_polars_and_pandas(
+    *,
+    data: st.DataObject,
+    time_series: pl.Series,
+    pl_every: str,
+    pd_alias: str,
+    pl_startby: StartBy,
+    closed: Literal["left", "right"],
+    number: int,
+) -> None:
+    nrows = len(time_series)
+    values = pl.Series(
```

Comment:
@honno: Interesting; do you think it would be worth putting any labels into the underlying primitives themselves?

Reply:
> @honno: Interesting; do you think it would be worth putting any labels into the underlying primitives themselves?

I think I'm correct in saying that there's no real point putting labels inside of `@st.composite`-functions, as Hypothesis would only report the overall thing been drawn at the test level, i.e. `test_foo(data): data.draw(dataframes(...), label="only the final df is reported")`. Labeling seems more useful anyway when a bespoke thing is for a test method.

Also, it's really cool that these primitives exist!

Reply:
Looking at the tests I think a few more native params might be useful too; some generic min_value & max_value args, for example... Don't hesitate to request anything you think would be a time-saver / useful :)

Reply:
not sure I see what you mean exactly, could you clarify what you're suggesting I change please?

Reply:
Oh! That’s more of a note to myself than you, sorry… Just thinking I can extend the primitives a bit to further simplify common cases 🤣